relay-metrics/src/aggregation.rs

```diff
@@ -745,6 +745,19 @@ impl BucketKey {
         std::hash::Hash::hash(self, &mut hasher);
         hasher.finalize() as i64
     }
+
+    fn validate_name(self, aggregator_config: &AggregatorConfig) -> Result<Self, ()> {
+        if self.metric_name.len() as u64 > aggregator_config.max_name_length {
+            relay_log::error!(
+                "Invalid metric name, too large ({} > max:{}): {:?}",
+                self.metric_name.len(),
+                aggregator_config.max_name_length,
+                self.metric_name
+            );
+            return Err(());
```

Comment:
We must apply the same validation to all the strings in `tags` (both key and values).